src/nvim/syntax.c

```diff
@@ -6874,8 +6874,20 @@ highlight_color (
   else if (!(TOLOWER_ASC(what[0]) == 'b' && TOLOWER_ASC(what[1]) == 'g'))
     return NULL;
   if (modec == 'g') {
-    if (fg)
+    if (what[2] == '#') {
+      if (fg) {
+          n = HL_TABLE()[id - 1].sg_rgb_fg;
+      } else if (sp) {
+          n = HL_TABLE()[id - 1].sg_rgb_sp;
+      } else {
+          n = HL_TABLE()[id - 1].sg_rgb_bg;
+      }
+      snprintf((char *)name, sizeof(name), "#%06x", n);
```

Comment:
Vim does something like this:

    snprintf(name, sizeof(name), "#%02x%02x%02x", (unsigned)(n >> 16), 
            (unsigned)(n >> 8) & 255, (unsigned)(n & 255))

Can you add a few more tests with values other than `#000` ?